### PR TITLE
IGAPP-309: Parallelize CI

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

**Comment by klinzo** [comment]
Another big CI review, sorry for that :/ Buuuuuut: This hopefully decreases the CI time quite a bit (once the large macos machine is unlocked, waiting for circleci customer support here).

Also: This actually removes some code! And I think it got a lot more clear :)

**Comment by klinzo** [comment]
# Overview over workflows
## Commit
image
## Commit (main)
image
## Native development delivery workflows
image
## Production delivery workflows
image

File: .circleci/src/jobs/notify_android.yml [hunk 1]
```diff
@@ -12,13 +12,10 @@ steps:
       at: .
   - restore_environment_variables
   - restore_yarn_cache
-  - run:
-      name: Create github release notes
-      command: echo "export GITHUB_RELEASE_NOTES=$(tools/parse-release-notes parse --android --language en)" >> ${BASH_ENV}
```
Comment (Taggotty) [inline .circleci/src/jobs/notify_android.yml:17]: why did you remove this part?

File: .circleci/src/workflows/scheduled_native_development_delivery.yml [hunk 2]
```diff
@@ -41,30 +52,71 @@ jobs:
         - tuerantuer-android
       production_delivery: false
       requires:
+        - check_native
         - e2e_android
+        - build_integreat_android
+
+#  - build_android:
```
Comment (Taggotty) [inline .circleci/src/workflows/scheduled_native_development_delivery.yml:59]: should there no malte development delivery in general or did you just comment this part out for now?

**Review by Taggotty — APPROVED** [review]

File: .circleci/src/jobs/notify_android.yml (hunk 1 above)
Comment (klinzo) [reply .circleci/src/jobs/notify_android.yml:17]: Somehow going the way with storing it in env vars breaks the newlines in the release notes.... I could not figure out why and how to fix it, so I decided to just execute it twice here....

File: .circleci/src/workflows/scheduled_native_development_delivery.yml (hunk 2 above)
Comment (klinzo) [reply .circleci/src/workflows/scheduled_native_development_delivery.yml:59]: Just for now. RIght now, we're still in the setup of the stores (or at least I am for ios). Once that is done, this has to be uncommented. There is an issue for this though: https://issues.tuerantuer.org/browse/IGAPP-253

E.g. the webapp delivery is done for malte as well.

**Review by maxammann — APPROVED** [review]
